python/dgl/distributed/dist_tensor.py

```diff
@@ -180,6 +181,22 @@ def __setitem__(self, idx, val):
         # TODO(zhengda) how do we want to support broadcast (e.g., G.ndata['h'][idx] = 1).
         self.kvstore.push(name=self._name, id_tensor=idx, data_tensor=val)
 
+    def __or__(self, other):
+        new_dist_tensor = self.__class__(
+                self._shape, self._dtype, part_policy=self._part_policy,
+                persistent=self._persistent, is_gdata=self._is_gdata,
+                attach=self._attach)
+        if not self.kvstore is other.kvstore is new_dist_tensor.kvstore:
+            raise DGLError('The underlying KVStores are not the same.')
```

Comment:
When will this happen?

Reply:
It should never happen.  Should I remove the check?

Reply:
Yes pls.